Issue with Snapshots and Asian Characters

**Describe the bug**
I just started getting an error with Syrupy today when adding a new integration test for an API endpoint I am working on.  I use Syrupy to diff the json returned vs a snapshot.  When updating the snapshot (since it doesn't exist yet) I get an error that seems to be related to the presence of asian characters.

```
snapshot = SnapshotAssertion(name='snapshot', num_executions=1)

    def test_get_mythic_raid_leaderboard(snapshot):
        client = WarcraftClient(os.getenv("CLIENT_ID"), os.getenv("CLIENT_SECRET"))
        response = client.get_mythic_raid_leaderboard("uldir", "alliance")
>       assert response == snapshot

src\tests\integration\test_warcraft_client.py:296:

.venv\lib\site-packages\syrupy\assertion.py:123: in __eq__
    return self._assert(other)
.venv\lib\site-packages\syrupy\assertion.py:138: in _assert
    self.extension.write_snapshot(
.venv\lib\site-packages\syrupy\extensions\base.py:146: in write_snapshot
    self._write_snapshot_fossil(snapshot_fossil=snapshot_fossil)
.venv\lib\site-packages\syrupy\extensions\amber.py:317: in _write_snapshot_fossil
    DataSerializer.write_file(snapshot_fossil_to_update)
.venv\lib\site-packages\syrupy\extensions\amber.py:46: in write_file
    f.write(f"{cls._indent}{data_line}")

self = <encodings.cp1252.IncrementalEncoder object at 0x00000219EF7B9C10>, input = "          'name': '阿尔法',\n", final = False

    def encode(self, input, final=False):
>       return codecs.charmap_encode(input,self.errors,encoding_table)[0]
E       UnicodeEncodeError: 'charmap' codec can't encode characters in position 19-21: character maps to <undefined>

..\..\AppData\Local\Programs\Python\Python38\lib\encodings\cp1252.py:19: UnicodeEncodeError
```

**To Reproduce**
Steps to reproduce the behavior:
1. Setup a "test" of an API that returns this JSON (https://gist.github.com/tehmufifnman/856ddd77aab073b9687aada15f4e8a6b)

```
def test_get_some_data(snapshot):
    client = SomeClient()
    response = client.get_some_data()
    assert response == snapshot
```

2. Run command 'pytest --snapshot-update'
3. See error

**Expected behavior**
I expect it to save a new initial snapshot.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Python Version: 3.8.2

--- a/src/syrupy/extensions/amber.py
+++ b/src/syrupy/extensions/amber.py
@@ -37,7 +37,7 @@ class DataSerializer:
         Writes the snapshot data into the snapshot file that be read later.
         """
         filepath = snapshot_fossil.location
-        with open(filepath, "w", newline="") as f:
+        with open(filepath, "w", encoding="utf-8", newline="") as f:
             for snapshot in sorted(snapshot_fossil, key=lambda s: s.name):
                 snapshot_data = str(snapshot.data)
                 if snapshot_data is not None:
@@ -57,7 +57,7 @@ class DataSerializer:
         indent_len = len(cls._indent)
         snapshot_fossil = SnapshotFossil(location=filepath)
         try:
-            with open(filepath, "r", newline="") as f:
+            with open(filepath, "r", encoding="utf-8", newline="") as f:
                 test_name = None
                 snapshot_data = ""
                 for line in f:

